### PR TITLE
ch4/ofi: Add CVAR to set a preferred NIC per rank

### PR DESCRIPTION
**Pull request by tarudoodi**
## Pull Request Description

The PR is based on top of https://github.com/pmodels/mpich/pull/6732.

Adds a `MPIR_CVAR_CH4_OFI_PREF_NIC` to let the user select a NIC per rank.
Also extends `MPIR_CVAR_DEBUG_SUMMARY` to print the NIC assigned to a rank.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [ ] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by tarudoodi** [comment]
test:mpich/ch4/ofi

**Comment by tarudoodi** [comment]
test:mpich/ch4/ofi

**Comment by tarudoodi** [comment]
test:mpich/ch4/ofi

**Comment by tarudoodi** [comment]
The ch3-tcp warnings test failed due to a cloning error: `ERROR: Error cloning remote repo 'origin'`

**Comment by tarudoodi** [comment]
@hzhou This is ready for review.

File: src/mpi/init/mpi_init.h
Comment (hzhou) [inline src/mpi/init/mpi_init.h:40]: You mean "Also print the preferred NIC for each rank"?

File: src/mpid/ch4/netmod/ofi/ofi_init.c
Comment (hzhou) [inline src/mpid/ch4/netmod/ofi/ofi_init.c:774]: How about `if (MPIR_CVAR_DEBUG_SUMMARY >= 2) {` ?

**Review by hzhou — COMMENTED** [review]
Please add commit messages for each commit.

**Comment by tarudoodi** [comment]
Hi @hzhou, I somehow missed the review messages. I agree with your comments. I'll push an update.

**Comment by tarudoodi** [comment]
> Please add commit messages for each commit.

Added.

File: src/mpi/init/mpi_init.h
Comment (tarudoodi) [reply src/mpi/init/mpi_init.h]: Done.

File: src/mpid/ch4/netmod/ofi/ofi_init.c
Comment (tarudoodi) [reply src/mpid/ch4/netmod/ofi/ofi_init.c]: Done.

**Comment by tarudoodi** [comment]
test:mpich/ch4/ofi

**Comment by tarudoodi** [comment]
test:mpich/ch4/ofi

**Comment by tarudoodi** [comment]
@hzhou The tests [passed](https://jenkins-pmrs.cels.anl.gov/job/mpich-review-ch4-ofi/4695/) and rebased the branch.

**Comment by hzhou** [comment]
image
Please do not merge `main` into branch. Rather, rebase on top of `main`.

**Review by hzhou — APPROVED** [review]
LGTM. Please rebase to `main`.

**Comment by tarudoodi** [comment]
> image Please do not merge `main` into branch. Rather, rebase on top of `main`.

Didn't expect the `update`  button to do that. Rebased on `main`. Is it ok to merge after basic warning checks pass?

**Comment by hzhou** [comment]
Yes, you can merge it

**Pull request merged**
